### PR TITLE
Initial Delivery to PaaS

### PR DESCRIPTION
**Pull request by sfawcett123**
# Purpose
Deliver Schools Experience to PaaS, ( review , development and Staging ) 

- Terraform
- Github actions

# Review Advise
A lot of the code is in master as it has been pushed along to work with the github actions. so try to ignore the change differences on this PR and accept all files are NEW

### CODE REVIEW

**Comment by dfe-schoolexperience** [comment]
Review can be found at https://se-review-1775-website.azurewebsites.net

**Comment by dfe-schoolexperience** [comment]
Review can be found at https://se-review-1775-website.azurewebsites.net

**Comment by dfe-schoolexperience** [comment]
Review can be found at https://se-review-1775-website.azurewebsites.net

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -494,3 +494,53 @@ jobs:
...
+          inputs: '{"environment": "Test", "sha": "${{ github.sha }}" }'
+          ref: ${{github.ref}}
+
+      - name: Wait for Deployment to Staging
+        uses: fountainhead/action-wait-for-check@v1.0.0
+        id: wait-for-deploy
+        with:
+          token: ${{ steps.azSecret.outputs.ACTIONS-API-ACCESS-TOKEN }}
+          checkName: Deploy Test
+          ref: ${{github.ref}}
```
Comment (saliceti) [inline .github/workflows/build.yml:533]: Shouldn't it be `github.sha` here? `github.ref` may change over time.

**Comment by dfe-schoolexperience** [comment]
Review can be found at https://se-review-1775-website.azurewebsites.net

File: terraform/paas/application.tf [hunk 2]
```diff
@@ -28,6 +28,14 @@ resource "cloudfoundry_app" "application" {
     route = cloudfoundry_route.route_internal.id
   }
 
+  dynamic "routes" {
+    for_each = data.cloudfoundry_route.app_route_internet
+    content {
+      route = routes.value["id"]
```
[the remark this answers is not in the document]
Comment (sfawcett123) [reply terraform/paas/application.tf:34]: The manual recommended this notation https://www.terraform.io/docs/language/expressions/dynamic-blocks.html so I went with the flow

File: terraform/paas/migrate/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+---
```
Comment (saliceti) [inline terraform/paas/migrate/manifest.yml:1]: Do we need this file in the repo?

File: terraform/paas/route.tf [hunk 4]
```diff
@@ -15,3 +21,4 @@ locals {
   app_endpoint = "${cloudfoundry_route.route_cloud.hostname}.${data.cloudfoundry_domain.cloudapps.name}"
 }
 
+
```
Comment (saliceti) [inline terraform/paas/route.tf:24]: Extra blank line?

File: terraform/paas/migrate/manifest.yml (hunk 3 above)
Comment (sfawcett123) [reply terraform/paas/migrate/manifest.yml:1]: Only while I am working on the code, once I am finished and have it all sorted I intend on deleting this.

File: terraform/paas/route.tf (hunk 4 above)
Comment (sfawcett123) [reply terraform/paas/route.tf:24]: Yes they were put in to force the files I had checked in earlier to look like they have changed, so it makes the review easier

File: .github/workflows/build.yml (hunk 1 above)
Comment (sfawcett123) [reply .github/workflows/build.yml:533]: It works on my Get Into Teaching, so if you could explain the difference I would be grateful

File: terraform/paas/test.env.tfvars [hunk 5]
```diff
@@ -0,0 +1,10 @@
+paas_space                = "get-into-teaching"
+paas_database_common_name = "school-experience-test-pg-common-svc"
+paas_redis_1_name         = "school-experience-test-redis-svc"
+paas_application_name     = "school-experience-app-test"
+paas_internet_hostnames   = ["staging-school-experience"]
```
Comment (saliceti) [inline terraform/paas/test.env.tfvars:5]: There is a bot of a mix of dev/test/staging. Can we keep just 1? I think this is the dev environment, right?

Comment (saliceti) [reply terraform/paas/test.env.tfvars:5]: See also test.bk.vars etc

Comment (sfawcett123) [reply terraform/paas/test.env.tfvars:5]: this is the test.env so is test, basically the confusion is I am changing test -> staging, so I will change this env file to staging and change the names

**Comment by dfe-schoolexperience** [comment]
Review can be found at https://se-review-1775-website.azurewebsites.net

File: .github/workflows/build.yml (hunk 1 above)
Comment (saliceti) [reply .github/workflows/build.yml:533]: We observed some pipeline failures with this setup. Leave it as it is, I’ll explain later.

**Review by felixtheflex — APPROVED** [review]

File: terraform/paas/application.tf (hunk 2 above)
[the remark this answers is not in the document]
Comment (felixtheflex) [reply terraform/paas/application.tf:34]: you can use both as the `routes` iteration object, has two attributes, `key` and `values`, which are of type maps - there attribute can be references as either routes.values.["attribute"] or  routes.values.["attribute"] or routes.values.attribute

**Review by saliceti — APPROVED** [review]

**Comment by dfe-schoolexperience** [comment]
Review can be found at https://se-review-1775-website.azurewebsites.net

**Pull request merged**